### PR TITLE
[Ide] Disable Active Runtime selection unless MONO_HACKER is set

### PR DESCRIPTION
**Pull request by jstedfast**
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/682117

### CODE REVIEW

**Comment by slluis** [comment]
We should use FeatureSwitchService for this. @rodrmoya can you explain/document how to use it?

**Comment by rodrmoya** [comment]
Yes, it's implemented here -> https://github.com/mono/monodevelop/blob/master/main/src/core/MonoDevelop.Core/MonoDevelop.Core.FeatureConfiguration/FeatureSwitchService.cs#L41

The idea is to have a single MD_FEATURES_ENABLED (and MD_FEATURES_DISABLED) env var that holds a (colon-separated) list of variable names, and, if needed, you can implement a IFeatureSwitchController to use more complex conditions (https://github.com/xamarin/md-addins/blob/master/Xamarin.Ide/Xamarin.Ide/VSMacFeatureSwitchController.cs).

In this case, I'd say you just need to call `FeatureSwitchService.IsFeatureEnabled ("MONO_HACKER")` (or `MonoHacker` to follow the naming other existing features use, although that's not important :D) instead of retrieving the env var yourself.

**Comment by jstedfast** [comment]
@rodrmoya thanks.

@slluis are you ok with MonoHacker? Or should we find some more descriptive feature name?

**Comment by slluis** [comment]
I would use RUNTIME_SELECTOR

**Comment by jstedfast** [comment]
Updated to use FeatureSwitchService and RUNTIME_SELECTOR

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Commands/ProjectCommands.cs [hunk 1]
```diff
@@ -504,7 +505,9 @@ internal class SelectActiveRuntimeHandler : CommandHandler
 	{
 		protected override void Update (CommandArrayInfo info)
 		{
-			if (IdeApp.Workspace.IsOpen && Runtime.SystemAssemblyService.GetTargetRuntimes ().Count () > 1) {
+			var enabled = FeatureSwitchService.IsFeatureEnabled ("RUNTIME_SELECTOR");
+
+			if (enabled.HasValue && enabled.Value && IdeApp.Workspace.IsOpen && Runtime.SystemAssemblyService.GetTargetRuntimes ().Count () > 1) {
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Commands/ProjectCommands.cs]: Nit:
```suggestion
			if (enabled.GetValueOrDefault() && IdeApp.Workspace.IsOpen && Runtime.SystemAssemblyService.GetTargetRuntimes ().Count () > 1) {
```

Comment (rodrmoya) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Commands/ProjectCommands.cs]: `if (enabled ?? false)` works too

Comment (jstedfast) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Commands/ProjectCommands.cs]: Just ended up using GetValueOrDefault(). Thanks!

**Comment by jstedfast** [comment]
Any update on this?

**Comment by jstedfast** [comment]
@slluis poke.

**Comment by slluis** [comment]
This is on Jordan now, see discussion in the work item.

**Comment by jstedfast** [comment]
This can be merged now

**Pull request closed without merging**